pub/lib/dependabot/pub/file_fetcher.rb

```diff
@@ -27,9 +27,12 @@ def fetch_files
         extra_pubspecs = Dir.glob("**/pubspec.yaml", base: clone_repo_contents)
         fetched_files += extra_pubspecs.map do |pubspec|
           relative_name = Pathname.new("/#{pubspec}").relative_path_from(directory)
-          fetch_file_from_host(relative_name)
+          file = fetch_file_from_host(relative_name)
```

Comment:
These are not files to be updated, but files to support resolution when there are relative path-dependencies.

I think instead of ignoring the symlink, we should probably read contents of the symlink target. Would that work?

Reply:
That works but I think it's basically the same result as filtering since the glob will also find the symlink it was pointed to (if it's in the project) and the `.uniq` will de-duplicate it. 

Unless there's something I missed which is always probable! I went ahead and pushed what I think you're saying, let me know if I got it right!

Reply:
Hmm, I'm not sure.
Given the following files
```
dir1/pubspec.yaml # real file
dir2/pubspec.yaml # symlink to ../dir1/pubspec.yaml
```
If I read your code correctly it will lead to the following files:

```
File(name: dir1/pubspec.yaml, content: read(dir1/pubspec.yaml))
File(name: dir1/pubspec.yaml, content: read(dir1/pubspec.yaml))
```
(which as you point out gets collapses by `uniq`)

We would like the following files:

```
File(name: dir1/pubspec.yaml, content: read(dir1/pubspec.yaml))
File(name: dir2/pubspec.yaml, content: read(dir1/pubspec.yaml))
```

That is for a symlink we want to preserve the original name, but the contents of the linked file.

In this way, if there is a dependency somewhere on `dir2/` it will resolve to a correct pubspec.

...Not sure of the best way to do this. :)